modules/apps/29-fee/types/metadata.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+// MetadataFromVersion attempts to parse the given string into a fee version Metadata,
+// an error is returned if it fails to do so.
+func MetadataFromVersion(version string) (Metadata, error) {
+	var metadata Metadata
+	err := ModuleCdc.UnmarshalJSON([]byte(version), &metadata)
+	if err != nil {
+		return Metadata{}, ErrInvalidVersion
```

Comment:
Should we indicate that the version could not be unmarshaled into the metadata?

Reply:
Originally was thinking that this would be wrapped by the caller like for example: `errorsmod.Wrapf(err, "failed to unmarshal ICS29 counterparty version metadata: %s)`

But I like this suggestion! Changed it to return this here: `errorsmod.Wrapf(ErrInvalidVersion, "failed to unmarshal metadata from version: %s", version)`. From caller we can just `Wrap(err, "invalid counterparty version")`